docs/source/architecture/index.rst

```diff
@@ -0,0 +1,122 @@
+
+Ordering Gate
+-------------
+
+It is an internal Iroha component (gRPC client) that relays `transactions <../core_concepts/glossary.html#transaction>`__ from `Peer Communication Service <#peer-communication-service>`__ to `Ordering Service <#ordering-service>`__.
+Ordering Gate recieves `proposals <../core_concepts/glossary.html#proposal>`_ (potential blocks in the chain) from Ordering Service and sends them to `Simulator <#simulator>`__ for `stateful validation <../core_concepts/glossary.html#stateful-validation>`__.
+It also requests proposal from the Ordering Service based on the consensus round. 
+
+Ordering Service
+----------------
+
+Internal Iroha component (gRPC server) that receives messages from other `peers <../core_concepts/glossary.html#peer>`__ and combines several `transactions <../core_concepts/glossary.html#transaction>`__ that have been passed `stateless validation <../core_concepts/glossary.html#stateless-validation>`__ into a `proposal <../core_concepts/glossary.html#proposal>`__. 
+Each node has its own ordering service.
+Proposal creation could be triggered by one of the following events:
+
+1. Time limit dedicated to transactions collection has expired.
+
+2. Ordering service has received the maximum amount of transactions allowed for a single proposal.
+
+Both parameters (timeout and maximum size of proposal) are configurable (check `environment-specific parameters <../guides/configuration.html#environment-specific-parameters>`_ page).
+
+A common precondition for both triggers is that at least one transaction should reach the ordering service.
+Otherwise, no proposal will be formed.
+
+Ordering service also performs preliminary validation of the proposals (e.g. clearing out statelessly rejected transactions from the proposal).
+
+Verified Proposal Creator
+-------------------------
+
+Internal Iroha component that performs `stateful validation <../core_concepts/glossary.html#stateful-validation>`_ of `transactions <../core_concepts/glossary.html#transaction>`__ contained in received `proposal <../core_concepts/glossary.html#proposal>`__ from the `Ordering Service <#ordering-service>`_.
+On the basis of transactions that have passed stateful validation **verified proposal** will be created and passed to `Block Creator <#block-creator>`__.
+All the transactions that have not passed stateful validation will be dropped and not included in a verified proposal.
+
+Block Creator
+-------------
+
+System component that forms a block from a set of transactions that have passed `stateless <../core_concepts/glossary.html#stateless-validation>`__ and `stateful <../core_concepts/glossary.html#stateful-validation>`__ validation for further propagation to `consensus <#consensus>`__.
+
+Block creator, together with the `Verified Proposal Creator <#verified-proposal-creator>`_ form a component called `Simulator <https://github.com/hyperledger/iroha/tree/master/irohad/simulator>`_.
+
+Consensus
```

Comment:
And rename the section to "Block Consensus (YAC)" to avoid confusion with the corresponding section in core concepts.

Reply:
done!